test/test.pl

```diff
@@ -2329,3 +2331,14 @@ sub test_reheader
              err=>'reheader/3_view1.sam.expected.err',
              cmd=>"$$opts{bin}/samtools reheader --in-place $$opts{path}/reheader/hdr.sam $fn.tmp.cram && $$opts{bin}/samtools view -h $fn.tmp.cram | perl -pe 's/\tVN:.*//'");
 }
+
+sub test_addrprg
+{
+    my ($opts,%args) = @_;
+
+    test_cmd($opts,out=>'addrprg/1_fixup.sam.expected', err=>'addrprg/1_fixup.sam.expected.err', cmd=>"$$opts{bin}/samtools addreplacerg -O sam -m overwrite_all $$opts{path}/addrprg/1_fixup.sam -", expect_fail=>0);
+    test_cmd($opts,out=>'addrprg/2_fixup_orphan.sam.expected', err=>'addrprg/2_fixup_orphan.sam.expected.err', cmd=>"$$opts{bin}/samtools addreplacerg -O sam -m orphan_only $$opts{path}/addrprg/2_fixup_orphan.sam -", expect_fail=>0);
+    test_cmd($opts,out=>'addrprg/1_fixup.sam.expected', err=>'addrprg/3_fixup.sam.expected.err', cmd=>"$$opts{bin}/samtools addreplacerg -O sam -R '1#7' $$opts{path}/addrprg/1_fixup.sam -", expect_fail=>1);
```

Comment:
Where's 3_fixup.sam.expected.err?  Why is this `expect_fail`?  (Should it perhaps be `want_fail=>1` instead?)